packages/protobuf-test/src/json.test.ts

```diff
@@ -1100,6 +1100,58 @@ describe("JSON parse errors", () => {
   }
 });
 
+describe("JSON serialize errors", () => {
+  const desc = proto3_ts.Proto3MessageDesc;
+  test("enum field not a number throws", () => {
+    const msg = create(desc);
+    // @ts-expect-error TS2322
+    msg.singularEnumField = "abc";
+    expect(() => toJson(desc, msg)).toThrow(
+      /^cannot encode enum spec.Proto3Enum to JSON: expected number, got "abc"$/,
+    );
```

Comment:
Before, this would raise an error with an empty message. Since users have run into this case a couple of times, it seems worth to have better error messages for this and similar cases.